packages/plugin-kit/src/fs/writeProjectFile.ts

```diff
@@ -0,0 +1,33 @@
+import * as fs from "node:fs/promises";
+import * as path from "node:path";
+
+import { SliceMachineHelpers } from "../createSliceMachineHelpers";
+
+export type WriteProjectFileArgs = {
+	filename: string;
+	contents: Parameters<typeof fs.writeFile>[1];
+	format?: boolean;
+	helpers: SliceMachineHelpers;
+};
+
+/**
+ * Writes a file to the project with optional formatting.
+ *
+ * @returns The file path to the written file.
+ */
+export async function writeProjectFile(
+	args: WriteProjectFileArgs,
+): Promise<string> {
+	const filePath = args.helpers.joinPathFromRoot(args.filename);
+
+	let contents = args.contents;
+
+	if (args.format) {
+		contents = await args.helpers.format(contents.toString(), filePath);
```

Comment:
What if we need to provide additional options to Prettier? For example I had(? not sure if I had to) to provide the Vue parser to format Vue files: https://github.com/prismicio/slice-machine/blob/c3a5117c0d35e86020df6b5d16dd600e4231fda3/packages/adapter-nuxt/src/hooks/project-init.ts#L144-L146

Could just be a `prettierOptions` optional argument 🤔

Reply:
Done! Added an optional `formatOptions` argument that accepts `format`'s third argument.